dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/webapp/WebApp.java

```diff
@@ -1055,6 +1059,11 @@ public void initialize(WebAppConfiguration config, DeployedModule moduleConfig,
                 }
             }
 
+         // if we're checkpointing, call commonInitializationFinally before the initTaskComplete
+            if (checkpointPhase == CheckpointPhase.APPLICATIONS) {
```

Comment:
Could you measure first response times for doing pingperf with `--at=deployment`?  I suspect we should be doing this early initialization for both `deployment` and `applications`.

Reply:
This change only helped the applications checkpoint option. Here are the numbers that I measured with and without the fix for each of the three options for two different applications The format is <startup>/<first request>:
crud applications checkpoint
fix: .105/.215
no fix: .106/.356
crud deployment checkpoint
fix: .362/.465
no fix: .224/.494
crud features:
fix: 1.003/1.100
no fix: .864/1.104
pingperf applications checkpoint
fix: .094/.131
no fix: .097/.255
pingperf deployment checkpoint
fix: .270/.300
no fix: .114/.286
pingperf features checkpoint
fix: .691/.720
no fix: .573/.714

Reply:
That is really surprising given that the pingperf app has no early startup code that gets executed as part of deployment.  For applications with no startup code I would expect `deployment` and `applications` to be exactly the same for checkpoint behavior.  We should investigate why we are not observing that here.